datatypes/src/raster/typed_raster.rs

```diff
@@ -91,96 +93,257 @@ where
         }
         None
     }
-}
 
-// TODO: use a macro?
-impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, u8, Vec<u8>>
-where
-    D: GridDimension,
-{
-    fn into(self) -> TypedRasterNDim<D> {
-        TypedRasterNDim::U8(self)
+    pub fn get_u8_ref(&self) -> Option<&BaseRaster<D, u8, Vec<u8>>> {
+        if let TypedRasterNDim::U8(r) = self {
+            return Some(r);
+        }
+        None
     }
-}
 
-impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, u16, Vec<u16>>
-where
-    D: GridDimension,
-{
-    fn into(self) -> TypedRasterNDim<D> {
-        TypedRasterNDim::U16(self)
+    pub fn get_u16_ref(&self) -> Option<&BaseRaster<D, u16, Vec<u16>>> {
+        if let TypedRasterNDim::U16(r) = self {
+            return Some(r);
+        }
+        None
     }
-}
 
-impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, u32, Vec<u32>>
-where
-    D: GridDimension,
-{
-    fn into(self) -> TypedRasterNDim<D> {
-        TypedRasterNDim::U32(self)
+    pub fn get_u32_ref(&self) -> Option<&BaseRaster<D, u32, Vec<u32>>> {
+        if let TypedRasterNDim::U32(r) = self {
+            return Some(r);
+        }
+        None
     }
-}
 
-impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, u64, Vec<u64>>
-where
-    D: GridDimension,
-{
-    fn into(self) -> TypedRasterNDim<D> {
-        TypedRasterNDim::U64(self)
+    pub fn get_u64_ref(&self) -> Option<&BaseRaster<D, u64, Vec<u64>>> {
+        if let TypedRasterNDim::U64(r) = self {
+            return Some(r);
+        }
+        None
     }
-}
 
-impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, i8, Vec<i8>>
-where
-    D: GridDimension,
-{
-    fn into(self) -> TypedRasterNDim<D> {
-        TypedRasterNDim::I8(self)
+    pub fn get_i8_ref(&self) -> Option<&BaseRaster<D, i8, Vec<i8>>> {
+        if let TypedRasterNDim::I8(r) = self {
+            return Some(r);
+        }
+        None
     }
-}
 
-impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, i16, Vec<i16>>
-where
-    D: GridDimension,
-{
-    fn into(self) -> TypedRasterNDim<D> {
-        TypedRasterNDim::I16(self)
+    pub fn get_i16_ref(&self) -> Option<&BaseRaster<D, i16, Vec<i16>>> {
+        if let TypedRasterNDim::I16(r) = self {
+            return Some(r);
+        }
+        None
     }
-}
 
-impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, i32, Vec<i32>>
-where
-    D: GridDimension,
-{
-    fn into(self) -> TypedRasterNDim<D> {
-        TypedRasterNDim::I32(self)
+    pub fn get_i32_ref(&self) -> Option<&BaseRaster<D, i32, Vec<i32>>> {
+        if let TypedRasterNDim::I32(r) = self {
+            return Some(r);
+        }
+        None
     }
-}
 
-impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, i64, Vec<i64>>
-where
-    D: GridDimension,
-{
-    fn into(self) -> TypedRasterNDim<D> {
-        TypedRasterNDim::I64(self)
+    pub fn get_i64_ref(&self) -> Option<&BaseRaster<D, i64, Vec<i64>>> {
```

Comment:
please sort them by size i8, i16, i32...

Reply:
they are sorted first by type i,u,f and then by size